src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -97,13 +99,19 @@ public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidCon
         if (fluidContainer.fluidType == null || fluidContainer.volume < fluidContainer.maxVolume) {
             getLiquidInReach(event.getInstigatorLocation(), event.getDirection(), 3).ifPresent(block -> {
                 EntityRef owner = item.getOwner();
+                EntityRef owner2 = owner.getOwner();
                 final EntityRef removedItem = inventoryManager.removeItem(owner, event.getInstigator(), item, false, 1);
                 //TODO: replace with better fluid handling, maybe by new CoreFluids module
+                FluidContainerItemComponent removedFluidContainer= removedItem.getComponent(FluidContainerItemComponent.class);
+                if(removedFluidContainer.fillingAmount == 0) {                                // not defined
+                    removedFluidContainer.fillingAmount = removedFluidContainer.maxVolume;
+                }
+                float volume= Math.min((removedFluidContainer.volume + removedFluidContainer.fillingAmount), removedFluidContainer.maxVolume);
+                DisplayNameComponent displayNameComponent = item.getComponent(DisplayNameComponent.class);
                 if (removedItem != null && block.isWater()) {
-                    // Set the contents of this fluid container and fill it up to max capacity.
-                    FluidUtils.setFluidForContainerItem(removedItem, "Fluid:Water",
-                            removedItem.getComponent(FluidContainerItemComponent.class).maxVolume);
-
+                    // Set the contents of this fluid container and fill it up to (current volume + filling amount).
+                    FluidUtils.setFluidForContainerItem(removedItem, "Fluid:Water", volume);
+                    owner2.send(new ChatMessageEvent( displayNameComponent.name + " filled to volume: "+ volume + " / " + removedFluidContainer.maxVolume + " ml.", owner2));
```

Comment:
Be aware that `displayNameComponent.name` may crash with a NPE if the entity does not have a DisplayNameComponent.